main.py

```diff
@@ -409,6 +416,62 @@ def check_arg_name(args):
         )
 
 
+def reload_model(args, ckpt_path, model, index, strict, task):
+    """
+        Function that reloads model if necesary and extracts trainable part 
+        of the model in preparation for target_task training. 
+        It only reloads model after the first task is trained, and only 
+        for finetuning. 
+
+        Parameters
```

Comment:
These types aren't that helpful, and if we're just going to include types, we should actually write them out in Python in the function signature. Instead, can we say what these params mean? Index is non-obvious.